app/scripts/services/dimStoreService.factory.js

```diff
@@ -1531,16 +1534,16 @@
       // year 1
 
       item.year = 1;
-      if (item.sourceHashes.includes(460228854) ||  // ttk
-          item.sourceHashes.includes(3523074641) || // variks
-          (item.talentGrid && item.talentGrid.infusable) || // no year1 item is infusable...
-          item.sourceHashes.includes(3739898362) || // elders challenge
-          item.sourceHashes.includes(3551688287)) { // kings fall
+      var infusable = (item.talentGrid && item.talentGrid.infusable);
+      if (infusable || _.intersection(yearHashes.year2, item.sourceHashes).length) {
         item.year = 2;
       }
-      if ((item.sourceHashes.includes(24296771) ||        // roi
-          !item.sourceHashes.length)) {                   // new items
-        item.year = 3;
+      if (!item.sourceHashes.length || _.intersection(yearHashes.year3, item.sourceHashes).length) {
```

Comment:
no because FoTL occured in both year2 and year3, the difference is one has tTK sourceHash as well